airflow/providers/google/cloud/hooks/cloud_run.py

```diff
@@ -0,0 +1,218 @@
+                                f'See details at {self._execution["status"]["logUri"]}.')
+            if verdict == 'False':
+                # This verdict means that a step failed,
+                # therefore the execution has failed
+                raise Exception(f'Cloud Run Job execution {self._exec_id} has failed. '
+                                f'See details at {self._execution["status"]["logUri"]}.')
+            if verdict == 'Unknown':
+                # This verdict means that the step is not completed yet,
+                # therefore the execution cannot be finished yet
+                return False
+            if verdict == 'True':
+                steps.append(step)
+        return CloudRunJobSteps.EXEC_STEP_COMPLETED in steps
+
+    def wait_for_done(self) -> None:
+        """Helper method to wait for result of job execution."""
+        self.log.info(f'Starting to poll status for Google Cloud Run job execution {self._exec_id}.')
+        self._execution = self._fetch_execution_by_id()
+        self.log.info(json.dumps(self._execution))
+        while not self._check_execution_state():
+            self.log.info(f'Waiting for execution completion. Sleeping {str(self._poll_sleep)} seconds.')
+            time.sleep(self._poll_sleep)
+            self._execution = self._fetch_execution_by_id()
+
+
+class CloudRunJobHook(GoogleBaseHook):
+    """
+    Hook for Google Cloud Run.
+
+    All the methods in the hook where project_id is used must be called with
+    keyword arguments rather than positional.
+    """
+
+    def __init__(
+        self,
+        gcp_conn_id: str = "google_cloud_default",
+        region: str = DEFAULT_CLOUD_RUN_REGION,
+        delegate_to: str | None = None,
+        poll_sleep: int = 10,
+        impersonation_chain: str | Sequence[str] | None = None,
+        delete_timeout: int | None = 5 * 60,
+        wait_until_finished: bool | None = None,
+        job_exec_cold_start: int | None = 10
+    ) -> None:
+        self.region = region
+        self.poll_sleep = poll_sleep
+        self.delete_timeout = delete_timeout
+        self.wait_until_finished = wait_until_finished
+        self.job_id: str | None = None
+        super().__init__(
+            gcp_conn_id=gcp_conn_id,
+            delegate_to=delegate_to,
+            impersonation_chain=impersonation_chain
+        )
+        self.job_exec_cold_start = job_exec_cold_start
+
+    def get_conn(self) -> build:
+        """Returns a Google Cloud Run service object."""
+        http_authorized = self._authorize()
+
+        # Use a regional endpoint since job execution is not possible from the global endpoint
+        client_options = ClientOptions(api_endpoint=f'https://{self.region}-run.googleapis.com')
+        return build('run', 'v1', client_options=client_options,
+                     http=http_authorized, cache_discovery=False)
```

Comment:
Why we are not using `JobsClient` from python SDK? https://cloud.google.com/python/docs/reference/run/latest/google.cloud.run_v2.services.jobs.JobsClient

Reply:
+1, the v2 API and its Python client libraries should be much easier to use.

Reply:
Hi @lwyszomi @steren 

Cannot install the python client library (google-cloud-run>=0.7.0) as it breaks other dependencies requirements.
Happy to use the JobsClient if we manage to install google-cloud-run library

Reply:
@Taragolis would you have an idea of how to fix this?

If using the official Python Client and implements all the CRUD operations on jobs are completion requirements for this PR, we should mark this PR as draft for the time being.

Reply:
FYI Completion requirements were discussed in this issue : https://github.com/apache/airflow/issues/24730.

Maybe we could consider all the Jobs CRUD operations in a later PR, when the official JobsClient can be installed properly?

Opened to suggestions here wink